### PR TITLE
Fixes issue #488. Using ENV variables for proxy

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty lmohanty@redhat.com

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
@praveenkumar I have tested this and works for me. 
@naina-verma can you please give it a spin on Windows

**Comment by praveenkumar** [comment]
@LalatenduMohanty It mean you don't have any variable in the Vagrantfile where you can let use edit those details and everytime they have to use env variable for proxy?

**Comment by LalatenduMohanty** [comment]
@praveenkumar Basically the idea is to give customer single way to give proxy information to CDK setup i.e. for both vagrant-registration and OpenShift. Also  I think we need to support environment variables. But if we can find someway to also provide the information through Vagrantfile that will be good.

**Comment by naina-verma** [comment]
@LalatenduMohanty  I tested it with hyperv and virtualbox providers , works fine !! ACK

File: components/rhel/rhel-ose/Vagrantfile [hunk 1]
```diff
@@ -64,6 +54,26 @@ Vagrant.configure(2) do |config|
     config.registration.password = ENV['SUB_PASSWORD']
   end
 
+  if ENV.has_key?('PROXY')
```
Comment (brgnepal) [inline components/rhel/rhel-ose/Vagrantfile]: You can try it in one line as :

```
  config.registration.proxy = PROXY = (ENV['PROXY'] || '')
  # other similar variables
```

In above case, lines from 73 to 75 are not required.

Comment (praveenkumar) [inline components/rhel/rhel-ose/Vagrantfile]: @budhrg So we can remove L#57-#75 by using below code right?

```
config.registration.proxy = PROXY = (ENV['PROXY'] || '')
config.registration.proxyUser = PROXY_USER = (ENV['PROXY_USER'] || '')
config.registration.proxyPassword = PROXY_PASSWORD = (ENV['PROXY_PASSWORD'] || '')
```

Comment (brgnepal) [inline components/rhel/rhel-ose/Vagrantfile]: Yes @praveenkumar . Well it should work and better to do it to be on safe side :smile:

**Comment by LalatenduMohanty** [comment]
@budhrg @praveenkumar I tested the new change and it is working fine.

**Comment by LalatenduMohanty** [comment]
I need to add the documentation change too

File: components/rhel/rhel-ose/Vagrantfile [hunk 2]
```diff
@@ -64,6 +54,11 @@ Vagrant.configure(2) do |config|
     config.registration.password = ENV['SUB_PASSWORD']
   end
 
+  #Proxy Information from environment
```
Comment (brgnepal) [inline components/rhel/rhel-ose/Vagrantfile]: Space after `#` is recommended.

**Comment by praveenkumar** [comment]
LGTM :+1:

**Pull request merged**